packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/eventbridge-scheduler/create-schedule.ts

```diff
@@ -0,0 +1,364 @@
+ */
+export interface RetryPolicy {
+  /**
+   * The maximum number of retry attempts to make before the request fails.
+   */
+  readonly maximumRetryAttempts: number;
+
+  /**
+   * The maximum amount of time to continue to make retry attempts.
+   */
+  readonly maximumEventAge: Duration;
+}
+
+/**
+ * Create a new AWS EventBridge Scheduler schedule
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/APIReference/API_CreateSchedule.html
+ */
+export class EventBridgeSchedulerCreateScheduleTask extends sfn.TaskStateBase {
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    super(scope, id, props);
+
+    this.validateProps(props);
+
+    if (props.kmsKey) {
+      props.kmsKey.grantDecrypt(props.target.role);
+    }
+
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: [
+          Stack.of(this).formatArn({
+            service: 'scheduler',
+            resource: 'schedule',
+            resourceName: `${this.props.groupName ?? 'default'}/${this.props.scheduleName}`,
+          }),
+        ],
+        actions: [
+          'scheduler:CreateSchedule',
+        ],
+      }),
+      new iam.PolicyStatement({
+        resources: [props.target.role.roleArn],
+        actions: [
+          'iam:PassRole',
+        ],
+      }),
+    ];
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('aws-sdk:scheduler', 'createSchedule', this.integrationPattern),
+      Parameters: {
+        ActionAfterCompletion: this.props.actionAfterCompletion,
+        ClientToken: this.props.clientToken,
+        Description: this.props.description,
+        EndDate: this.props.endDate ? this.props.endDate.toISOString() : undefined,
+        FlexibleTimeWindow: {
+          Mode: this.props.flexibleTimeWindow ? 'FLEXIBLE' : 'OFF',
+          MaximumWindowInMinutes: this.props.flexibleTimeWindow?.toMinutes(),
+        },
+        GroupName: this.props.groupName,
+        KmsKeyArn: this.props.kmsKey?.keyArn,
+        Name: this.props.scheduleName,
+        ScheduleExpression: this.props.schedule.expressionString,
+        ScheduleExpressionTimezone: this.props.timezone,
+        StartDate: this.props.startDate ? this.props.startDate.toISOString() : undefined,
+        State: (this.props.enabled ?? true) ? 'ENABLED' : 'DISABLED',
+        Target: this.props.target.renderTargetObject(),
+      },
+    };
+  }
+
+  private validateProps(props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    if (props.startDate && props.endDate && props.startDate > props.endDate) {
```

Comment:
```suggestion
    if (props.startDate && props.endDate && props.startDate >= props.endDate) {
```
Can we specify equal dates?